src/shared/utilities/timeoutUtils.ts

```diff
@@ -128,20 +128,25 @@ export class Timeout {
     }
 }
 
+interface WaitUntilOptions {
+    /** Timeout in ms (default: 5000) */
+    readonly timeout?: number
+    /** Interval in ms between fn() checks (default: 500) */
+    readonly interval?: number
+    /** Wait for "truthy" result, else wait for any defined result including `false` (default: true) */
+    readonly truthy?: boolean
+}
+
 /**
  * Invokes `fn()` until it returns a non-undefined value.
  *
  * @param fn  Function whose result is checked
- * @param opt.timeout  Timeout in ms (default: 5000)
- * @param opt.interval  Interval in ms between fn() checks (default: 500)
- * @param opt.truthy  Wait for "truthy" result, else wait for any defined result including `false` (default: true)
+ * @param options  See {@link WaitUntilOptions}
  *
  * @returns Result of `fn()`, or `undefined` if timeout was reached.
  */
-export async function waitUntil<T>(
-    fn: () => Promise<T>,
-    opt: { timeout: number; interval: number; truthy: boolean } = { timeout: 5000, interval: 500, truthy: true }
-): Promise<T | undefined> {
+export async function waitUntil<T>(fn: () => Promise<T>, options: WaitUntilOptions): Promise<T | undefined> {
+    const opt = { timeout: 5000, interval: 500, truthy: true, ...options }
```

Comment:
is the nominal type actually needed or could we just add this line and change the param to:

    opt: { timeout?: number; interval?: number; truthy?: boolean }

Reply:
It's needed if you want documentation on the fields, otherwise it's not.
We have a few places where we use JSDoc to annotate but that doesn't really work.